service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CarePlanMapper.java

```diff
@@ -0,0 +1,131 @@
+    private final String SNOMED = "2.16.840.1.113883.2.1.3.2.4.15";
+    private final String INFORMATION_ADVICE_GIVEN = "1052951000000105";
+
+    private List<CarePlan> carePlansCreated = new ArrayList<>();
+
+    public List<CarePlan> mapCarePlan(POCDMT000002UK01ClinicalDocument1 clinicalDocument,
+                                      Encounter encounter) {
+
+        POCDMT000002UK01StructuredBody structuredBody = getStructuredBody(clinicalDocument);
+
+        return Arrays.stream(structuredBody.getComponentArray())
+                .map(POCDMT000002UK01Component3::getSection)
+                .map(this::findCarePlanSections)
+                .flatMap(List::stream)
+                .map(section -> createCarePlanFromSection(section, encounter))
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public CarePlan createCarePlanFromSection(POCDMT000002UK01Section cpSection, Encounter encounter) {
+        CarePlan carePlan = new CarePlan();
+        carePlan.setIdElement(newRandomUuid());
+        carePlan
+                .setIntent(PLAN)
+                .setSubject(encounter.getSubject())
+                .setSubjectTarget(encounter.getSubjectTarget())
+                .setStatus(ACTIVE)
+                .setContextTarget(encounter)
+                .setContext(new Reference(encounter))
+                .setPeriod(encounter.getPeriod());
+
+        if (cpSection.isSetLanguageCode()) {
+            carePlan.setLanguage(NodeUtil.getNodeValueString(cpSection.getLanguageCode()));
+        }
+        if (cpSection.isSetTitle()) {
+            carePlan.setTitle(NodeUtil.getNodeValueString(cpSection.getTitle()));
+        }
+
+        if (cpSection.getText().sizeOfContentArray() > 0) {
+            String cpTextContent = NodeUtil.getNodeValueString(cpSection.getText().getContentArray(0));
+            Narrative narrative = new Narrative();
+            narrative.setDivAsString(cpTextContent);
+            narrative.setStatus(Narrative.NarrativeStatus.GENERATED);
+            if (cpSection.isSetText()) {
+                carePlan.setText(narrative);
+            }
+            carePlan.setDescription(cpTextContent);
+        }
+
+        return carePlan;
+    }
+
+    private POCDMT000002UK01StructuredBody getStructuredBody(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
+        return Optional.ofNullable(clinicalDocument)
+                .map(POCDMT000002UK01ClinicalDocument1::getComponent)
+                .map(POCDMT000002UK01Component2::getStructuredBody)
+                .orElse(null);
+    }
+
+    @SafeVarargs
+    public static Predicate<DomainResource> ofTypes(Class<? extends DomainResource>... types) {
+        return Arrays.stream(types)
+                .map(type -> (Predicate<DomainResource>) type::isInstance)
+                .reduce(Predicate::or)
+                .orElse(x -> false);
+
+    }
+
+    private List<POCDMT000002UK01Section> findCarePlanSections(POCDMT000002UK01Section section) {
+
```

Comment:
Please run code reform on all files you edit / create

Reply:
Updated